sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1343,15 +1343,7 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils with Te
         .collect()
     }.getMessage
     assert(msg.contains("only include the internal corrupt record column"))
-    intercept[org.apache.spark.sql.catalyst.errors.TreeNodeException[_]] {
-      spark
-        .read
-        .option("columnNameOfCorruptRecord", columnNameOfCorruptRecord)
-        .schema(schema)
-        .csv(testFile(valueMalformedFile))
-        .filter($"_corrupt_record".isNotNull)
-        .count()
-    }
```

Comment:
@gengliangwang . Any comments about this test coverage removal?

Reply:
Hm, yea. Looks 1. this test had to throw `AnalysisException`, 2. this test should not be removed as of  https://github.com/apache/spark/pull/24005/files#diff-3ab796758cd96a99cce4ab12bc80c12cR63

Reply:
For V2 will still throw `AnalysisException`, while in V1 it throws `TreeNodeException`. Should we test both behaviors? Is testing `.collect()` enough?

Reply:
I think both tests were added because both cases were documented (https://github.com/apache/spark/pull/24005/files#diff-3ab796758cd96a99cce4ab12bc80c12cR63), so it at least had a reason.

If it's troublesome for some reasons to fix this test, we can remove it too for now. Technically this test looks not correctly testing the case.